core/trino-main/src/test/java/io/trino/sql/planner/TestUnwrapCastInComparison.java

```diff
@@ -815,20 +818,32 @@ private void testRemoveFilter(String inputType, String inputPredicate)
         }
     }
 
+    private void testUnwrap(String inputType, String inputPredicate)
+    {
+        testUnwrap(getQueryRunner().getDefaultSession(), inputType, inputPredicate, Optional.empty());
+    }
+
     private void testUnwrap(String inputType, String inputPredicate, String expectedPredicate)
     {
-        testUnwrap(getQueryRunner().getDefaultSession(), inputType, inputPredicate, expectedPredicate);
+        testUnwrap(getQueryRunner().getDefaultSession(), inputType, inputPredicate, Optional.of(expectedPredicate));
     }
 
     private void testUnwrap(Session session, String inputType, String inputPredicate, String expectedPredicate)
+    {
+        testUnwrap(session, inputType, inputPredicate, Optional.of(expectedPredicate));
+    }
+
+    private void testUnwrap(Session session, String inputType, String inputPredicate, Optional<String> expectedPredicate)
     {
         String sql = format("SELECT * FROM (VALUES CAST(NULL AS %s)) t(a) WHERE %s", inputType, inputPredicate);
         try {
+            PlanMatchPattern values = values("a");
             assertPlan(sql,
                     session,
                     output(
-                            filter(expectedPredicate,
-                                    values("a"))));
+                            expectedPredicate
+                                    .map(predicate -> filter(predicate, values))
+                                    .orElse(values)));
```

Comment:
We should assert  that the _other_ values have no rows (pruned away). Otherwise the test seems to be showing that filter is just gone with the wind.

Reply:
Done